### ISSUE TITLE
"Run job later" schedules are lost when rundeck instance restarts

### ISSUE THREAD

**Issue by S0LERA** [rundeck/rundeck#8327]
**Describe the bug**
I have a rundeck instance in a container where I schedule executions of some jobs with the "Run Job Later" option, the problem is that if I restart that rundeck instance, the schedules are lost. This is a problem, since a containerized instance must support being rebooted and redeployed without losing work.

I don't know if these schedules are saved in the system's internal crontab, but they should be saved as a database entry, to prevent them from being lost if the rundeck instance is restarted.

**My Rundeck detail**

* Rundeck version: [4.11.0]
* install type: [docker]
* OS Name/version: [-]
* DB Type/version: [postgres]

**To Reproduce**
1. Go to some job
2. Click on "Run Job Later..."
3. Restart rundeck
4. The schedule is lost

**Expected behavior**
The schedules are maintained even if the rundeck instance is restarted.

**Desktop (please complete the following information):**
 - OS: [Windows 10]
 - Browser [Mozilla Firefox]
 - Version [107.0]

**Comment by UnicodeTreason** [comment]
Check the service.log, in my ancient v2 Rundeck any Scheduled Jobs that don't rely on a manually entered secure option survive a restart. You can see them being recreated in the Service log.

**Comment by S0LERA** [comment]
Yes, i just check that and the only schedules lost are the ones with secure options. But I need this schedules with secure options. What is the approach to solve this?

**Comment by UnicodeTreason** [comment]
I haven't had a chance to investigate it personally, I just ensure no secure option jobs are scheduled the day the VM get patched.
I have a hunch that maybe using a saved key instead of a manually entered option might work.

**Comment by elwinh** [comment]
Have you set the UUID to a specific value to survive a reboot as per:
https://github.com/rundeck/rundeck/issues/4181?

**Comment by S0LERA** [comment]
> Have you set the UUID to a specific value to survive a reboot as per: #4181?

Hello,

We just try setting the UUID but no changes, jobs with "secure" options don't survive to reboot.

**Comment by UnicodeTreason** [comment]
Have you had a go at pulling the secure options from the internal Key Storage instead of user input?
I have yet to get the time to test it as a solution personally.

![Image](https://github.com/user-attachments/assets/c7904054-5ec2-476c-9b7c-65f12024a8e1)

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
